libraries/app/application.cpp

```diff
@@ -1020,6 +1023,10 @@ void application::set_program_options(boost::program_options::options_descriptio
           "For asset_api::get_asset_holders to set its default limit value as 100")
 		   ("api-limit-get-key-references",boost::program_options::value<uint64_t>()->default_value(100),
 		    "For database_api_impl::get_key_references to set its default limit value as 100")
+         ("api-limit-get-htlc-by",boost::program_options::value<uint64_t>()->default_value(100),
+          "For database_api_impl::get_htlc_by_from and get_htlc_by_to to set its default limit value as 100")
+         ("api-limit-get-full-accounts",boost::program_options::value<uint64_t>()->default_value(100),
```

Comment:
IMO the option name is confusing. The other limits apply to the number of list entries returned by the call. Here, the number of accounts is unlimited, but the setting limits the number of entries inside each account's sub-lists.

IMO we should limit the number of accounts that can be queried in one go as well. This is an easy DOS vector.

Reply:
fixed this by having 2 limit optiosn for get_full_accounts: `api-limit-get-full-accounts` for the account vector and `api-limit-get-full-accounts-lists` for the list results.

Reply:
Thanks. @sschiessl-bcp does the UI call `get_full_accounts` with many accounts at once? How many typically?

Reply:
Actually the default is currently 10 accounts but i forgot to change the 100 here, https://github.com/bitshares/bitshares-core/blob/3aba3f76d47e6e4b20707c69a51de43d5dc3f84b/libraries/app/include/graphene/app/application.hpp#L51
Still, will wait for @sschiessl-bcp response before fixing this and changing in the other places if needed.

Reply:
`bitsharesjs` only uses one account at all times, so does UI (at least up what I could find)

Reply:
Speaking of: How hard is it to add pagination, or limit to `get_key_references` and `get_account_references`?

There have been attacks on several accounts by creating dummy accounts and adding keys of the account to attack.

Reply:
@sschiessl-bcp i opened a new ticket to discuss and implement limit and pagination for the mentioned api calls: https://github.com/bitshares/bitshares-core/issues/1753

what do we think about the 10 limit by default for `get_full_accounts`?

Reply:
@oxarbitrage: @sschiessl-bcp has mentioned elsewhere that the reference wallet only fetches one account every time with `get_full_accounts`.

Reply:
yes, i agree with 10.